test/serverExplorer.test.ts

```diff
@@ -0,0 +1,193 @@
+import * as chai from 'chai';
+import * as sinon from 'sinon';
+import * as sinonChai from 'sinon-chai';
+import { ServersViewTreeDataProvider } from '../src/serverExplorer';
+import { RSPClient, Protocol } from 'rsp-client';
+import { EventEmitter, window } from 'vscode';
+
+const expect = chai.expect;
+chai.use(sinonChai);
+
+suite('Server explorer', () => {
+
+    let sandbox: sinon.SinonSandbox;
+    const clientStub: sinon.SinonStubbedInstance<RSPClient> = new RSPClient('somehost', 8080);
+    let serverExplorer: ServersViewTreeDataProvider;
+
+    setup(() => {
+        serverExplorer = new ServersViewTreeDataProvider(clientStub);
+        sandbox = sinon.createSandbox();
+    });
+
+    teardown(() => {
+        sandbox.restore();
+    });
+
+    const serverType: Protocol.ServerType = {
+        description: 'a type',
+        id: 'type',
+        visibleName: 'the type'
+    };
+
+    const serverHandle: Protocol.ServerHandle = {
+        id: 'id',
+        type: serverType
+    };
+
+    const stateChange: Protocol.ServerStateChange = {
+        server: serverHandle,
+        state: 1
+    };
+
+    const ProcessOutput: Protocol.ServerProcessOutput = {
+        processId: 'process id',
+        server: serverHandle,
+        streamType: 0,
+        text: 'the type'
+    };
+
+    const findServerBeans = {
+        length: 1,
+        fullVersion: 'version',
+        location: 'path',
+        name: 'EAP',
+        serverAdapterTypeId: 'org.jboss',
+        specificType: 'EAP',
+        typeCategory: 'EAP',
+        version: '7.1'
+    };
+
+    const status = {
+        code: 0,
+        message: 'ok',
+        pluginId: 'unknown',
+        severity: 0
+    };
+
+    test('InsertServer call should add server to tree data model', () => {
+        const refreshStub = sandbox.stub(serverExplorer, 'refresh').returns(null);
+        const getChildren = serverExplorer.getChildren(null);
+        serverExplorer.insertServer(serverHandle);
+        expect(refreshStub).calledOnce;
+        expect(getChildren).deep.equals(undefined);
```

Comment:
This is wrong. You adding she server and then your checking if the server model **does not have** server you just added. What you should test that the `getChildren` call returns array with one element and that element is the `ServerHandler` you passed as a parameter to `insertServer`. 

You should write test that actually help you verify code is working, not just make test pass.